### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by tamnnm** [dniloash/Pywinter#14]
When running pywinter on 20-Century Reanalysis data, I get the error Segmentation Fault for all Soil, 2D, 3D variables. As I trace back, it's mostly likely due to the Fortran file. However, I cannot pinpoint where that comes from.

The dimension of my data is only (31 levels, 361 longitude,181 latitude) which is quite coarse. Therefore, I wonder what this could come from? 

I attach below the code I have implemented: 
[pywinter_code.txt](https://github.com/user-attachments/files/17433760/pywinter_code.txt)

**Comment by dniloash** [comment]
Hello, about this topic.
I obtained a file (air.2m.1881.nc) from this source (https://www.psl.noaa.gov/data/gridded/data.20thC_ReanV3.html) and was able to create files without the problem you describe. This could be a problem associated with your system but I am not sure.
About the attached code. I don't see any problems at a glance, but I can't really test without the data. Also I would be careful with the R_levels variable, I think the database uses Milibars and the V3dp function needs Pa.

https://github.com/dniloash/Pywinter/blob/master/examples/user_ex/example_user01.ipynb

**Comment by tamnnm** [comment]
Hi, 
The data I have retrieved is from the same source that you have mentioned. 
Since that day, I have tried modified the Fortran file, and found out that it fails at opening the '.inTer1' without knowing exactly why. 
I have used iostat to investigate the matter without success.
Since I have use the os.chdir in the main python to run, can this be a problem at any chance? Nevertheless, I have tested the working directly of both the .so file and the winter.py stays the same where I wanted. 
I will get back when verified it again.

Thank you for your time.
